test/integration/fuzzing_test.lua

```diff
@@ -0,0 +1,2687 @@
+            {{{'list', Nullable, k, Nullable, {v.value}, Nullable, k, NonNullable, box.NULL}}, nil},
+            -- (31) Argument: [T] -> Variable: [T!] -> Value: [] -> Default: [value(s)] - OK
+            {{{'list', Nullable, k, Nullable, {}, Nullable, k, NonNullable, {v.default}}}, nil},
+            -- (32) Argument: [T] -> Variable: [T!] -> Value: [] -> Default: [] - OK
+            {{{'list', Nullable, k, Nullable, {}, Nullable, k, NonNullable, {nil}}}, nil},
+            -- (33) Argument: [T] -> Variable: [T!] -> Value: [] -> Default: [null] - OK
+            {{{'list', Nullable, k, Nullable, {}, Nullable, k, NonNullable, {box.NULL}}}, nil},
+            -- (34) Argument: [T] -> Variable: [T!] -> Value: [] -> Default: nil - OK
+            {{{'list', Nullable, k, Nullable, {}, Nullable, k, NonNullable, nil}}, nil},
+            -- (35) Argument: [T] -> Variable: [T!] -> Value: [] -> Default: null - OK
+            {{{'list', Nullable, k, Nullable, {}, Nullable, k, NonNullable, box.NULL}}, nil},
+            -- (36) Argument: [T] -> Variable: [T!] -> Value: [null] -> Default: [value(s)] - FAIL
+            {
+                {{'list', Nullable, k, Nullable, {box.NULL}, Nullable, k, NonNullable, {v.default}}},
+                'Variable \"var1[1]\" expected to be non-null',
+            },
+            -- (37) Argument: [T] -> Variable: [T!] -> Value: [null] -> Default: [] - FAIL
+            {
+                {{'list', Nullable, k, Nullable, {box.NULL}, Nullable, k, NonNullable, {nil}}},
+                'Variable \"var1[1]\" expected to be non-null',
+            },
+            -- (38) Argument: [T] -> Variable: [T!] -> Value: [null] -> Default: [null] - FAIL
+            {
+                {{'list', Nullable, k, Nullable, {box.NULL}, Nullable, k, NonNullable, {box.NULL}}},
+                'Variable \"var1[1]\" expected to be non-null',
+            },
+            -- (39) Argument: [T] -> Variable: [T!] -> Value: [null] -> Default: nil - FAIL
+            {
+                {{'list', Nullable, k, Nullable, {box.NULL}, Nullable, k, NonNullable, nil}},
+                'Variable \"var1[1]\" expected to be non-null',
+            },
+            -- (40) Argument: [T] -> Variable: [T!] -> Value: [null] -> Default: null - FAIL
+            {
+                {{'list', Nullable, k, Nullable, {box.NULL}, Nullable, k, NonNullable, box.NULL}},
+                'Variable \"var1[1]\" expected to be non-null',
+            },
+            -- (41) Argument: [T] -> Variable: [T!] -> Value: nil -> Default: [value(s)] - OK
+            {{{'list', Nullable, k, Nullable, nil, Nullable, k, NonNullable, {v.default}}}, nil},
+            -- (42) Argument: [T] -> Variable: [T!] -> Value: nil -> Default: [] - OK
+            {{{'list', Nullable, k, Nullable, nil, Nullable, k, NonNullable, {nil}}}, nil},
+            -- (43) Argument: [T] -> Variable: [T!] -> Value: nil -> Default: [null] - OK
```

Comment:
Why it's not an error? `box.NULL` is not a valid `NonNullable` `k`.

Reply:
This is acrroding to http://spec.graphql.org/October2021/#sec-Combining-List-and-Non-Null